trin-storage/src/versioned/mod.rs

```diff
@@ -0,0 +1,31 @@
+pub mod sql;
+pub mod store;
+mod utils;
+
+use rusqlite::types::{FromSql, FromSqlError, ValueRef};
+use strum::{AsRefStr, Display, EnumIter, EnumString};
+
+pub use store::VersionedContentStore;
+pub use utils::create_store;
+
+#[derive(Copy, Clone, Debug, Display, Eq, PartialEq, EnumString, AsRefStr, EnumIter)]
+#[strum(serialize_all = "snake_case")]
+pub enum ContentType {
+    Beacon,
+    History,
+    State,
+}
+
+#[derive(Copy, Clone, Debug, Display, Eq, PartialEq, EnumString, AsRefStr, EnumIter)]
+#[strum(serialize_all = "snake_case")]
+pub enum StoreVersion {
+    LegacyContentData,
+    IdIndexed,
```

Comment:
I'm not sure I follow the vision here, if you can help me understand. For each new "version" we would add an entry to the `StoreVersion` enum? Is this better than just using numbers or semver(might be overkill)? If we just use these "strings" to identify each version...
- it is helpful to understand the changes included in the update (though, this can be accomplished with comments)
- it's not intuitive to understand the order of the upgrades (though, this can be accomplished with comments)
I'm not sure, it's just seems like using something like `v0`, `v1` .. `v100` is a bit more intuitive